src/components/post/PostHighlights.vue

```diff
@@ -0,0 +1,240 @@
+        <div class="cursor-pointer" @click="onClose">
+          <i class="fa fa-lg fa-close"></i>
+        </div>
+      </div>
+      <div ref="modal-body" class="modal-body">
+
+        <comment-highlight
+          v-if="canComment"
+          :forumThreadId="threadId"
+          :highlight="highlight"
+          :autoFocus="true" />
+
+        <comments-list
+          :filter="'highlight'"
+          :post="post"
+          :forumThreadId="threadId"
+          :commentCount="commentCount"
+          :comments="commentsSorted" />
+
+      </div>
+      <div class="modal-footer">
+        <button class="btn-close" @click="onClose">
+          Close
+        </button>
+      </div>
+    </div>
+    <div class="modal-overlay" @click="onClose" />
+  </div>
+</template>
+
+<script>
+import clone from 'lodash/clone'
+import isEqual from 'lodash/isEqual'
+import CommentHighlight from '@/components/comment/CommentHighlight'
+import CommentsList from '@/components/comment/CommentsList'
+
+export default {
+  props: {
+    post: {
+      type: Object
+    },
+    comments: {
+      type: Array,
+      default: () => [],
+    },
+    commentCount: {
+      type: Number,
+      default: 0,
+    },
+    isDone: {
+      type: Boolean,
+      default: false,
+    },
+    highlight: {
+      type: String,
+      defaults: '',
+    }
+  },
+
+  components: {
+    CommentsList,
+    CommentHighlight,
+  },
+
+  data () {
+    return {
+      isActive: false,
+      commentId: null,
+      threadId: null,
+    }
+  },
+
+  watch: {
+    $route (to, from) {
+      const isActive = (to.query.thread_id || to.query.comment_id)
+      const element = this.$el.querySelector(`[data-selector="c${to.query.comment_id}"]`)
+      const modalBodyEl = this.$refs['modal-body']
+
+      if (!isEqual(to.query, from.query)) {
+        this.isActive = isActive
+        this.commentId = to.query.comment_id
+        this.threadId = to.query.thread_id
+      }
+
+      if (element) {
+        return this.$nextTick(() => {
+          // Reset scroll position
+          modalBodyEl.scrollTo(0, 0)
+
+          const offset = 20
+          const modalBodyTop = modalBodyEl.getBoundingClientRect().top
+          const scrollTop = Math.round(element.getBoundingClientRect().top || 0)
+
+          modalBodyEl.scrollTo(0, scrollTop - modalBodyTop - offset)
+        })
+      }
+
+      this.$nextTick(() => modalBodyEl.scrollTo(0, 0))
+    },
+
+    isActive(value) {
+      const scrollTop = window.scrollY
+
+      document.body.style.overflow = value ? 'hidden' : ''
+
+      requestAnimationFrame(() => {
+        return window.scrollTo(0, scrollTop)
+      }, 1)
+    },
+  },
+
+  computed: {
+    canComment () {
+      return (!this.comments.filter(c => c.highlight === this.highlight).length)
+    },
+
+    commentsSorted () {
+      return clone(this.comments).sort((a, b) => {
+        return new Date(b.dateCreated) - new Date(a.dateCreated)
```

Comment:
Is this date really a `string`? For me, in something similar, this just works:

```javascript
return o1.dateCreated >= o2.dateCreated ? -1 : 1;
```

Reply:
Yes, this date is a string.